private/dependency_tree_parser.bzl

```diff
@@ -323,7 +350,21 @@ def _generate_imports(repository_ctx, dep_tree, explicit_artifacts, neverlink_ar
             all_imports.append("alias(\n\tname = \"%s\",\n\tactual = \"%s\",\n%s)" %
                                (versioned_target_alias_label, target_label, alias_visibility))
 
-            # 11. If using maven_install.json, use a genrule to copy the file from the http_file
+            # 13. If there is a corresponding for the artifact, create a license target.
+            #
+            # license(
```

Comment:
I am a little confused here. Why do we need to append both `applicable_licenses` and `License` target? I thought one is sufficient.

Reply:
`applicable_licenses` is what adds the `license` instance to the artifact, so the license metadata can be retrieved from the providers when the aspect is traversing the dependency graph.

The reason we need to define the `license` instance in the build file is because it is not defined anywhere else. We are passing in the metadata with JSON, not the actual `license` instance, so the metadata needs to be assembled here to define the `license`. The reason the `license_kind` is not declared here is because `license_kinds` are re-usable across many `license`s and the developer is either using the pre-defined `license_kinds` in the `rules_license` repo or their orgs' internal `license_kind` definitions.